logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt

```diff
@@ -182,8 +185,47 @@ class ConversationDataSource(
      * Gets a flow that allows observing of
      */
     override suspend fun observeConversationDetailsById(conversationID: ConversationId): Flow<ConversationDetails> =
-        conversationDAO.observeGetConversationByQualifiedID(idMapper.toDaoModel(conversationID)).wrapStorageRequest().onlyRight()
-            .map(conversationMapper::fromDaoModel).flatMapLatest(::getConversationDetailsFlow)
+        conversationDAO.observeGetConversationByQualifiedID(idMapper.toDaoModel(conversationID))
+            .wrapStorageRequest()
+            .onlyRight()
+            .map(conversationMapper::fromDaoModel)
+            .flatMapLatest(::getConversationDetailsFlow)
+
+    private suspend fun getConversationDetailsFlow(conversation: Conversation): Flow<ConversationDetails> = when (conversation.type) {
+        Conversation.Type.SELF -> flowOf(ConversationDetails.Self(conversation))
+        // TODO(user-metadata): get actual legal hold status
+        Conversation.Type.GROUP -> flowOf(
```

Comment:
as far as we have `flowOf` here, I assume it won't "update" every time when a new message comes into conversation. 
But I guess it should, isn't it? Or did I get something wrong?

Reply:
This function will not indeed but the previous one calling it in the chain does work correctly  and updates whenever field in the conversation updates